orchagent/neighorch.cpp

```diff
@@ -574,6 +574,19 @@ void NeighOrch::doTask(Consumer &consumer)
 
         IpAddress ip_address(key.substr(found+1));
 
+        /* Verify Ipv4 LinkLocal and skip neighbor entry added for RFC5549 */
+        if ((ip_address.getAddrScope() == IpAddress::LINK_SCOPE) && (ip_address.isV4()))
+        {
+            /* Check if this prefix is not a configured ip, if so allow */
+            IpPrefix ipll_prefix(ip_address.getV4Addr(), 16);
+            if (!m_intfsOrch->isPrefixSubnet (ipll_prefix, alias))
```

Comment:
Can't we ignore ipv4 LLA specifically? what happens if we receive the nbr update from kernel after removing the IP address on an interface?

Reply:
This check is added to allow neighbors if the l3 interface has the same prefix.
Like if user creates a interface using v4LL range address (config interface ip add Ethernet46 169.254.0.10/24), it should be allowed.

Reply:
if ipv6 is enabled, can't we block this configuration in CLI?

Reply:
I think, it is not appropriate to block it.

Reply:
We should make sure we create only V4 LLA as NH not as a neighbor to point to any route.

Reply:
I think this can be handled separately.